pkg/engine/lifeycletest/pulumi_test.go

```diff
@@ -1476,26 +1476,6 @@ func TestAliases(t *testing.T) {
 		},
 	}}, []deploy.StepOp{deploy.OpSame, deploy.OpReplace, deploy.OpCreateReplacement, deploy.OpDeleteReplaced})
 
-	// ensure failure when different resources use duplicate aliases
```

Comment:
This should still test ok now?

Reply:
This is an error we can no longer detect. The setup for this test is... odd. It starts from an empty base state and wants to validate that two resources cannot mention the same alias. The only way to do that from the statefile alone is to always store all of the aliases in the statefile. In order to support this check, we'd need to move the handling of that specific case into the step generator. I did this in a different branch, but I got a little bit more ambitious there and fixed a few other bugs I found along the way: https://github.com/pulumi/pulumi/compare/pgavlin/choose-single-alias

IMO, this specific check is relatively low-value, as it's really only covering the case where resources that _have no base state_ cannot be mention the same alias. If the resources are both aliased to the same base state, then we will still issue an error (just added a test for that).

There is one interesting case that we don't catch in either case: given the base state `[A]`, it is possible to register a resource `A` and a resource `B` that is aliased to `A`. This is bad, and we should fix it, but that requires fixing a nasty bug in provider aliasing. That bug is an easy fix, I think (a possible fix is included in that other branch), but I'd like to separate it out nonetheless.